nsq-mixin/dashboards/topics.libsonnet

```diff
@@ -0,0 +1,239 @@
+local grafana = import 'github.com/grafana/grafonnet-lib/grafonnet/grafana.libsonnet';
+local prometheus = grafana.prometheus;
+local nsqgrafonnet = import '../lib/nsqgrafonnet/nsqgrafonnet.libsonnet';
+
+{
+
+  grafanaDashboards+:: {
+
+    local nsqSelector = 'job=~"$job", instance=~"$instance"',
+    local nsqTopicSelector = nsqSelector + ',topic=~"$topic"',
+    local nsqChannelSelector = nsqTopicSelector + ',channel=~"$channel"',
+    local nsqTopicDataLinkQueryParams = '${datasource:queryparam}&${job:queryparam}&$${__url_time_range}',
+    local nsqChannelDataLinkQueryParams = '${datasource:queryparam}&${job:queryparam}&${__url_time_range}',
+
+    local nsqTopics =
+      nsqgrafonnet.timeseries.new('Topic messages')
+      .addTarget(prometheus.target(expr='avg by (topic) (rate(nsq_topic_message_count{%s}[$__rate_interval]))' % nsqTopicSelector, intervalFactor=1, legendFormat='{{topic}} rps'))
+      .addTarget(prometheus.target(expr='sum by () (rate(nsq_topic_message_bytes{%s}[$__rate_interval]))' % nsqTopicSelector, intervalFactor=1, legendFormat='selected topics Bps rate'))
```

Comment:
To my eye, this graph get's a bit difficult to understand, particularly with the mixed units.

If I understand this correctly, the left Y axis is rate of topics per second, while the right Y axis is the Bps rate of select queries?

Does this make more sense as two timeseries? Maybe a gauge for current select Bps, and a pie/donut graph for busiest topic?

Reply:
I know what you mean, but let's keep it as is for now. I made Bps shady as this metric is more for 'style'.

Reply:
I'm good with this. We'll see if we get feedback from the field ;)